packages/main/src/MultiComboBox.js

```diff
@@ -402,6 +404,12 @@ class MultiComboBox extends UI5Element {
 		});
 
 		this.fireEvent("selectionChange", { items: this._getSelectedItems() });
+
+		if (isEnter(event.detail) || (!event.detail.selectionComponentPressed && !isSpace(event.detail))) {
```

Comment:
I think you can skip this, when the checkbox is on focus and Enter is pressed, selectionComponentPressed is still false.
Both conditions would be true

isEnter(event.detail) - > true
!event.detail.selectionComponentPressed && !isSpace(event.detail)) - true